src/pencilkit.cs

```diff
@@ -246,9 +297,168 @@ interface PKToolPicker {
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("inkType")]
+		NSString WeakInkType { get; }
+
+		[Wrap ("PKInkTypeExtensions.GetValue(WeakInkType)")]
+		PKInkType InkType { get; }
+
+		[Export ("color")]
+		UIColor Color { get; }
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface PKFloatRange : NSCopying
+	{
+		[Export ("initWithLowerBound:upperBound:")]
+		IntPtr Constructor (nfloat lowerBound, nfloat upperBound);
+
+		[Export ("lowerBound")]
+		nfloat LowerBound { get; }
+
+		[Export ("upperBound")]
+		nfloat UpperBound { get; }
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface PKStroke : NSCopying
+	{
+		[Export ("initWithInk:strokePath:transform:mask:")]
+		IntPtr Constructor (PKInk ink, PKStrokePath path, CGAffineTransform transform, [NullAllowed] BezierPath mask);
+
+		[Export ("ink")]
+		PKInk Ink { get; }
+
+		[Export ("transform")]
+		CGAffineTransform Transform { get; }
+
+		[Export ("path")]
+		PKStrokePath Path { get; }
+
+		[NullAllowed, Export ("mask")]
+		BezierPath Mask { get; }
+
+		[Export ("renderBounds")]
+		CGRect RenderBounds { get; }
+
+		[Export ("maskedPathRanges")]
+		PKFloatRange[] MaskedPathRanges { get; }
+	}
+
+	delegate void PKInterpolatedPointsEnumeratorHandler (PKStrokePoint strokePoint, out bool stop);
+
+	[Mac (10, 16), iOS (14, 0)]
+	[Introduced (PlatformName.MacCatalyst, 14, 0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface PKStrokePath : NSCopying
+	{
+		[Export ("initWithControlPoints:creationDate:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (PKStrokePoint[] controlPoints, NSDate creationDate);
+
+		[Export ("count")]
+		nuint Count { get; }
+
+		[Export ("creationDate")]
+		NSDate CreationDate { get; }
+
+		[Export ("pointAtIndex:")]
+		PKStrokePoint GetPoint (nuint index);
+
+		[Export ("objectAtIndexedSubscript:")]
+		PKStrokePoint GetObject (nuint indexedSubscript);
+
+		[Export ("interpolatedLocationAt:")]
+		CGPoint GetInterpolatedLocation (nfloat parametricValue);
+
+		[Export ("interpolatedPointAt:")]
+		PKStrokePoint GetInterpolatedPoint (nfloat parametricValue);
+
+		[Export ("enumerateInterpolatedPointsInRange:strideByDistance:usingBlock:")]
+		void EnumerateInterpolatedPointsByDistance (PKFloatRange range, nfloat distanceStep, PKInterpolatedPointsEnumeratorHandler enumeratorHandler);
+
+		[Export ("enumerateInterpolatedPointsInRange:strideByTime:usingBlock:")]
+		void EnumerateInterpolatedPointsByTime (PKFloatRange range, double timeStep, PKInterpolatedPointsEnumeratorHandler enumeratorHandler);
+
+		[Export ("enumerateInterpolatedPointsInRange:strideByParametricStep:usingBlock:")]
+		void EnumerateInterpolatedPointsByParametric (PKFloatRange range, nfloat parametricStep, PKInterpolatedPointsEnumeratorHandler enumeratorHandler);
```

Comment:
Just an idea, but would this be better?

```suggestion
		void EnumerateInterpolatedPointsByParametricStep (PKFloatRange range, nfloat parametricStep, PKInterpolatedPointsEnumeratorHandler enumeratorHandler);
```

It sounds slightly better to me, but depending on how the API is used one or the other would make more sense.

In any case, probably not a big deal.

Reply:
I like this. According to the documentation, `Executes a given block using each point in a range with a parametric step.` + same for all of these other `EnumerateInterpolated...` methods here. I think it makes sense given the use to change the name here.